src/Engines/QueryBuilderEngine.php

```diff
@@ -340,10 +341,13 @@ protected function compileRelationSearch($query, $relation, $column, $keyword)
                 $relationMethod = "whereRaw";
             }
 
+            //overwrite the first element of the binding - we expect the keyword to be located there
+            $bindings[0]  = $this->prepareKeyword($keyword);
```

Comment:
I find this to be a bit of a hack. What if we have some where statement from the base query?